bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsActions.java

```diff
@@ -54,10 +54,15 @@ public void setThingHandler(@Nullable ThingHandler handler) {
     private void triggerSwitch(String variableName) {
         try {
             if (handler != null) {
+
                 handler.writeValue(variableName, "1");
             }
         } catch (HeliosException e) {
             logger.warn("Error executing action triggering switch for variable {}: {}", variableName, e.getMessage());
+        } catch (InterruptedException e) {
+            logger.warn(
+                    "{} encountered Exception when trying to lock Semaphore for writing variable {} to the device: {}",
+                    HeliosEasyControlsHandler.class.getSimpleName(), variableName, e.getMessage());
```

Comment:
Your changes look good! As a shut-down of OH is a normal operation, you could make this debug or trace. It's even ok to not log it at all, but simply finish the program flow as your do it already.